### PR TITLE
fix coordination of fs instance: resolves #149

### LINKED ISSUE #144: Add abstract decorator for Storage
It is inconvenient to create implementations of Storage composed of other implementations without an abstract decorator class

### LINKED ISSUE #149: Fix coordination of fs instance in FileStorage class

### PR DESCRIPTION
**Pull request by Guseyn**
fix coordination of fs instance: resolves #149

### CODE REVIEW

**Comment by Guseyn** [comment]
@g4s8 plz check this pr.

**Comment by 0crat** [comment]
This pull request [#150](https://github.com/artipie/asto/issues/150) is assigned to @sammers21[/z](https://www.0crat.com/u/sammers21), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/5f9d9c9e-96ab-4001-af7a-7412e703f925); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; @sammers21[/z](https://www.0crat.com/u/sammers21) is not a member of this project yet, but they can request to join, as [§1](http://www.zerocracy.com/policy.html#1) explains; we should be aware that @sammers21[/z](https://www.0crat.com/u/sammers21) is on vacation; this ticket may be delayed; there will be no monetary reward for this job

File: src/main/java/com/artipie/asto/fs/FileStorage.java [hunk 1]
```diff
@@ -47,6 +47,11 @@
 /**
  * Simple storage, in files.
  *
+ * <p>
+ * This class must be constructed only via vertx instance, otherwise we can get file system from
+ * another vertx instance and it will cause performance issues.
```
Comment (Sammers21) [inline src/main/java/com/artipie/asto/fs/FileStorage.java:52]: Does the passing of another vert.x instance would not cause the mentioned "performance issues"?

**Comment by Sammers21** [comment]
@Guseyn,  what is "coordination of fs instance"?

**Comment by 0crat** [comment]
Job was finished in 23 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @sammers21[/z](https://www.0crat.com/u/sammers21), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Pull request [#150](https://github.com/artipie/asto/issues/150) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by sereshqua** [comment]
@Sammers21 please make sure you will find at least 3 issues during next CR + please also make sure you start all your comments with the name of the users they are referred to, [see](https://github.com/artipie/asto/pull/150#discussion_r408571211)

@Guseyn please make sure all problems found by the code reviewer (@Sammers21 ) are always addressed by the pull request author (you), which means that they were either fixed as the reviewer wanted or the reviewer was convinced that they should not be fixed;

**Comment by sereshqua** [comment]
@Sammers21 @Guseyn ping

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request closed without merging**